hack/setup.sh

```diff
@@ -61,19 +78,36 @@ main() {
                         install_kpack
                         install_knative_serving
                         install_tekton
+                        install_tekton_git_cli_task
                         ;;
 
                 example)
-                        setup_example
-                        test_example
+                        test_runnable_example
+                        teardown_runnable_example
+
+                        setup_example_sc "basic-sc"
+                        test_example_sc "basic-sc"
+                        teardown_example_sc "basic-sc"
+
+                        setup_example_sc "testing-sc"
+                        test_example_sc "testing-sc"
+                        teardown_example_sc "testing-sc"
+
+                        test_gitops
+
+                        log "all tests passed!!"
                         ;;
 
                 teardown-example)
-                        teardown_example
+                        teardown_runnable_example
+                        teardown_example_sc "basic-sc"
+                        teardown_example_sc "testing-sc"
+                        teardown_gitops_example
```

Comment:
`teardown_gitops_example` is already called in `test_gitops` so I think it'd end up being called twice if we leave this here. Maybe it should only be here and we shouldn't teardown in `test_gitops`

Reply:
This is true for the other tests as well. If they succeed, they clean up after themselves. If everything has succeeded, teardown-example needn't be called.

So why have any of the tests clean up? Shortcut to avoid test pollution.